pybossa/cache/projects.py

```diff
@@ -125,13 +125,32 @@ def format_task(row, lock_users=[]):
 
     if filter_user_prefs:
         # construct task list for worker view
+
+        # TODO: may need to refactor code once Task Browse page optimization is settled
+        sql = """ SELECT task.id,
+                (
+                    SELECT CAST(COUNT(id) AS FLOAT) as ct FROM task_run
+                    WHERE task_id = task.id
+                ) as n_task_runs,
+                task.n_answers,
+                (
+                    SELECT MAX(finish_time) as ft FROM task_run
+                    WHERE task_id = task.id
+                ) as ft,
+                priority_0,
+                task.created,
+                task.calibration,
+                task.user_pref,
+                task.worker_filter,
+                task.worker_pref
+                FROM task
+                WHERE task.project_id =:project_id""" + filters
+
```

Comment:
so we now have two different queries for task browse and task list. any reason/risk for task browse query [here](https://github.com/bloomberg/pybossa/pull/697/files/d7010cfd47d97dc0987e5028cf67f8995a5293ca#diff-f9ab7f32f66f66784ac342b1f9576deb5a06260956ae9386553a73796e4ebbaeR109-R120) not to have similar optimization as task list?

Reply:
Jeff and I tested that applying similar pattern to task browse doesn't improve the perf as it does on task list feature.